DSEGOG-15 adding support for icons for columns

sam-glendenning opened the pull request:
## Description
I've added a small range of icons for columns based on mapping the column header name to a specific Material icon. Due to the limited test data we currently have, as well as the huge range of columns we can have, it's hard to define icons for more than the ID, shot number and timestamp columns, which we will always receive data on so can be confident it will appear in the display.

For any remaining columns, I currently show a default `FeedIcon` as it's vague enough to do the job for now. The mapping approach is flexible enough that icons can be easily added or changed later, it just relies on a mapping existing for the `dataKey` of the specified column.

## Column Icons
ID: 
![image](https://user-images.githubusercontent.com/71134574/177587282-f23190c7-6a9a-4695-bf1e-1183ccb3b6bc.png)

Shot Number:
![image](https://user-images.githubusercontent.com/71134574/177587443-ddad684c-4d55-4fec-bc13-173b49b7a977.png)

Timestamp:
![image](https://user-images.githubusercontent.com/71134574/177587568-24f54479-6c28-49f3-a194-588288a837db.png)

Default icon (FeedIcon):
![image](https://user-images.githubusercontent.com/71134574/177587715-8d59bdfa-7991-48fc-8d54-5556919193a2.png)

## Testing instructions
Note the icons used for ID, shot number, timestamp and channel columns and give thoughts.

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking
Closes DSEGOG-15

Commit:
DSEGOG-15 adding three column icons plus one default icon

I've mapped three columns (ID, shot number and timestamp) to individual column icons, as I know this data will always be present in the API response. As we can't be sure what other data will come back yet, or what format it takes, I've left out any icons for that for now. However, I have assigned a default FeedIcon for anything that isn't one of those three

Commit:
DSEGOG-15 fixing grammatical mistake

louise-davies commented on src/table/headerRenderers/dataHeader.component.tsx:
```diff
@@ -81,14 +83,21 @@ const DataHeader = (props: DataHeaderProps): React.ReactElement => {
       onClick={() => onSort(dataKey, nextSortDirection)}
     >
-      <Typography noWrap sx={{ fontSize: 'inherit', lineHeight: 'inherit' }}>
+      {columnIconMappings.get(dataKey.toUpperCase()) ?? <FeedIcon />}
```
If we're defining the Icons like this (aka via a global map & lookup via `dataKey`) - then isn't the `icon` prop obsolete? If so we should remove the code relating to the `icon` prop method.

sam-glendenning replied:
That was my bad for missing that... see my latest commit, which uses the correct method. Keep an eye on snapshot changes

Commit:
DSEGOG-15 rejigging icons passed as props, fixing sorting label

Now making use of the existing icon method, with icons passed as props from table instead. The placement of the icons in this method also means that clicking an icon won't accidentally sort the column either

Commit:
Merge branch 'develop' into DSEGOG-15-icons-for-columns

louise-davies reviewed (approved):
Looks good - we'll just have to revisit this later once we know the source of Icon info (unless just using the `Feed` icon is fine)
